### PR TITLE
Cache last scan results

### PR DESCRIPTION
**Pull request by asafgabai**
- [x] All [tests](https://github.com/jfrog/ide-plugins-common/actions/workflows/test.yml) passed. If this feature is not already covered by the tests, I added new tests.
-----
Save last scan results in cache and show them on plugin startup.

### CODE REVIEW

File: src/main/java/com/jfrog/ide/common/nodes/IssueNode.java [hunk 1]
```diff
@@ -1,11 +1,23 @@
...
+import java.util.UUID;
 
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY,
+        getterVisibility = JsonAutoDetect.Visibility.NONE,
+        setterVisibility = JsonAutoDetect.Visibility.NONE)
+@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "uuid")
 public abstract class IssueNode extends DefaultMutableTreeNode implements SubtitledTreeNode, Comparable<IssueNode> {
+    // For deserialization
+    @SuppressWarnings("unused")
+    private String uuid = UUID.randomUUID().toString();
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/common/nodes/IssueNode.java:19]:
```suggestion
    private final String uuid = UUID.randomUUID().toString();
```

File: src/main/java/com/jfrog/ide/common/persistency/ScanCache.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import java.util.List;
+
+import static com.jfrog.ide.common.utils.Utils.createMapper;
+
+/**
+ * Cache for Xray scan results.
+ *
+ * @author yahavi
+ */
+public class ScanCache {
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/common/persistency/ScanCache.java:27]: This class deserves a comprehensive unit tests

File: src/main/java/com/jfrog/ide/common/persistency/ScanCacheObject.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+public class ScanCacheObject {
+    static int CACHE_VERSION = 4;
+
+    @JsonProperty("version")
+    int version = CACHE_VERSION;
+
+    @JsonProperty("fileTreeNodes")
+    List<FileTreeNode> fileTreeNodes;
+    @JsonProperty("scanTime")
+    long scanTime;
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/common/persistency/ScanCacheObject.java:23]: Isn't it a scan timestamp?
```suggestion
    @JsonProperty("scanTimestamp")
    long scanTimestamp;
```

File: src/main/java/com/jfrog/ide/common/persistency/ScanCache.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    private final File file;
+    private ObjectMapper objectMapper;
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/common/persistency/ScanCache.java:29]: Can be initialized inline in the declaration:
```suggestion
    private final ObjectMapper objectMapper = createMapper();
```

**Review by yahavi — CHANGES_REQUESTED** [review]
Well done, @asafgabai!

File: src/main/java/com/jfrog/ide/common/persistency/ScanCache.java (hunk 2 above)
Comment (asafgabai) [reply src/main/java/com/jfrog/ide/common/persistency/ScanCache.java:27]: You're right. They'll be added separately.

**Pull request merged**